lib/rpc/globalRpcProviders.ts

```diff
@@ -0,0 +1,62 @@
+import { ChainId } from '@uniswap/sdk-core'
+import SingleJsonRpcProvider from './singleJsonRpcProvider'
+import UniJsonRpcProvider from './uniJsonRpcProvider'
+
+export default class GlobalRpcProviders {
+  private static readonly PROVIDER_RPC_URL_RANKING: Partial<Record<ChainId, number[] | undefined>> = {
+    [ChainId.MAINNET]: undefined,
+  }
+
+  private static readonly PROVIDER_RPC_URL_WEIGHTS: Partial<Record<ChainId, number[] | undefined>> = {
+    [ChainId.MAINNET]: undefined,
+  }
+
+  private static SINGLE_RPC_PROVIDERS: Partial<Record<ChainId, SingleJsonRpcProvider[]>> | null = null
+
+  private static UNI_RPC_PROVIDERS: Partial<Record<ChainId, UniJsonRpcProvider>> | null = null
+
+  private static initGlobalSingleRpcProviders() {
+    const INFURA_KEY = process.env.UNI_RPC_PROVIDER_INFURA_KEY
+    if (INFURA_KEY === undefined) {
+      throw new Error(`UNI_RPC_PROVIDER_INFURA_KEY must be a defined environment variable`)
+    }
+    const QUICKNODE_MAINNET_RPC_URL = process.env.UNI_RPC_PROVIDER_QUICKNODE_MAINNET_RPC_URL
+    if (QUICKNODE_MAINNET_RPC_URL === 'undefined') {
+      throw new Error(`REACT_APP_QUICKNODE_MAINNET_RPC_URL must be a defined environment variable`)
+    }
+    this.SINGLE_RPC_PROVIDERS = {
```

Comment:
This implementation will make the RPC provider gateway to roll out to ETH Mainnet completely. I think we want to de-risk by deploying to a chain with lesser usages/revenues, meanwhile we know we are using the Infura RPC for that chain.

By looking at the Infura RPCs, and the chains we collect revenue on, Avalanche 43114 might be a good choice.